### PR TITLE
common : relax sampler name matching

### PR DESCRIPTION
**Pull request by ddh0**
## Overview

Currently, in some cases, the alternative names for samplers (like `top-k` and `min-p` instead of the canonical `top_k` and `min_p`) are not always recognized by the `common_sampler_types_from_names` function in `common/sampling.cpp`.

This PR changes the signature of this function to remove the `bool allow_alt_names` flag, and removes all occurences of the flag from call sites. Therefore, the function will now always match all known names.

This fixes an issue I was seeing wherein samplers specified in the `llama-server` UI were not recognized as valid when the alternative names were used:

```console
193.02.179.393 W common_sampler_types_from_names: unable to match sampler by name 'top-k'
193.02.179.399 W common_sampler_types_from_names: unable to match sampler by name 'min-p'
193.02.179.400 W common_sampler_types_from_names: unable to match sampler by name 'temp'
```

## Additional information

I also changed the logic of the function to unconditionally check the provided sampler names against both the canonical and alternative names, and to be case-insensitive. Finally, I added a couple more aliases such as `topk` and `topp` to cover all the bases.

## Requirements

- I have read and agree with the [contributing guidelines](https://github.com/ggml-org/llama.cpp/blob/master/CONTRIBUTING.md)
- AI usage disclosure: NO

### CODE REVIEW

**Comment by ddh0** [comment]
I can confirm that this fixes the problem I was having with setting samplers from inside the `llama-server` UI:

```console
114.35.392.776 I slot launch_slot_: id  0 | task -1 | sampler chain: logits -> top-k -> top-p -> min-p -> ?temp-ext -> dist
```

**Review by ServeurpersoCom — APPROVED** [review]
Tested locally.

Hyphenated names that were rejected before now resolve fine:

  sampler chain: logits -> top-k -> top-p -> min-p -> ?temp-ext -> dist

generation_settings echo confirms: ["top-k","top-p","min-p","temp"] -> ["top_k","top_p","min_p","temperature"],
case insensitivity and aliases work, and unknown names are still dropped with a warning. 

No CLI regression: --samplers was already permissive, and the UI round-trips canonical names from the server, so the default flow is unchanged. The only behavior delta is that previously-rejected alt names now resolve.

LGTM.

**Comment by ddh0** [comment]
cc @CISC, could I please get another review/approval? (or, who should I ping in this case? thank you.)

**Comment by ddh0** [comment]
Actually @ngxson, I remember seeing that you were the last to touch a lot of this code, so I don't want to change anything unexpected.

File: common/sampling.cpp [hunk 1]
```diff
@@ -776,39 +776,49 @@ std::vector<common_sampler_type> common_sampler_types_from_names(const std::vect
...
             continue;
         }
-        if (allow_alt_names) {
-            sampler = sampler_alt_name_map.find(name);
-            if (sampler != sampler_alt_name_map.end()) {
-                samplers.push_back(sampler->second);
-                continue;
-            }
+        // check alternative names
+        sampler = sampler_alt_name_map.find(name_lower);
```
Comment (CISC) [inline common/sampling.cpp]: Since it's no longer optional, why keep the separate maps and double lookup, just put them all in one (put alternates at the end and rename the map).

**Review by CISC — APPROVED** [review]

File: common/sampling.cpp (hunk 1 above)
Comment (ddh0) [reply common/sampling.cpp]: Thanks, addressed in 5f6fe7a

**Comment by ddh0** [comment]
I think it's good to go. Re-approval, please?

**Comment by CISC** [comment]
> I think it's good to go. Re-approval, please?

I would have preferred canonical names at the top and alternates commented and separate last, but let's have @ngxson decide. :)

**Comment by ddh0** [comment]
I've separated the canonical vs. alternative sampler names within the single mapping now as CISC asked, and merged in the latest from `master`. cc @ngxson could you please take a look when you have a chance? thank you :)

File: common/sampling.cpp [hunk 2]
```diff
@@ -761,54 +761,55 @@ std::string common_sampler_type_to_str(enum common_sampler_type cnstr) {
...
     samplers.reserve(names.size());
 
     for (const auto & name : names) {
-        auto sampler = sampler_canonical_name_map.find(name);
-        if (sampler != sampler_canonical_name_map.end()) {
+        // transform the given sampler name to lowercase before checking
+        std::string name_lower = name;
+        std::transform(name.begin(), name.end(), name_lower.begin(), [](char c){
+            return std::tolower(static_cast<unsigned char>(c));
+        });
```
Comment (CISC) [inline common/sampling.cpp]:
```suggestion
        std::transform(name_lower.begin(), name_lower.end(), name_lower.begin(), ::tolower);
```

**Review by CISC — APPROVED** [review]

**Review by CISC — APPROVED** [review]

**Review by pwilkin — APPROVED** [review]

**Review by ngxson — COMMENTED** [review]
just wondering, but can we have `sampler_name_map` in format `kebab-case`, then have the alias to be autogenerated? For example, we seem to have only 2 rules now:
- `-` to `_`
- `-` to `` (removed)

The only exception is `typ-p` --> `typ`

**Comment by ddh0** [comment]
> just wondering, but can we have `sampler_name_map` in format `kebab-case`, then have the alias to be autogenerated? For example, we seem to have only 2 rules now:
> - `-` to `_`
> - `-` to `` (removed)
> 
> The only exception is `typ-p` --> `typ`

Addressed now in [`e12bbc5`](https://github.com/ggml-org/llama.cpp/pull/23744/commits/e12bbc53f2fa04c4545806cc8f9fa9822c2044e2), let me know if this is what you had in mind.

**Comment by CISC** [comment]
> > just wondering, but can we have `sampler_name_map` in format `kebab-case`, then have the alias to be autogenerated? For example, we seem to have only 2 rules now:
> > 
> > * `-` to `_`
> > * `-` to `` (removed)
> > 
> > The only exception is `typ-p` --> `typ`
> 
> Addressed now in [`e12bbc5`](https://github.com/ggml-org/llama.cpp/pull/23744/commits/e12bbc53f2fa04c4545806cc8f9fa9822c2044e2), let me know if this is what you had in mind.

Just `merge` the alt map into the main map so you don't have to do two lookups.

File: common/sampling.cpp [hunk 3]
```diff
@@ -769,54 +769,65 @@ std::string common_sampler_type_to_str(enum common_sampler_type cnstr) {
...
+            }
+        }
+
+        // misc. aliases
+        m.insert({"nucleus", COMMON_SAMPLER_TYPE_TOP_P});
+        m.insert({"temp",    COMMON_SAMPLER_TYPE_TEMPERATURE});
+        m.insert({"typ",     COMMON_SAMPLER_TYPE_TYPICAL_P});
+
+        // merge with the canonical map
+        m.insert(canonical_name_map.begin(), canonical_name_map.end());
```
Comment (CISC) [inline common/sampling.cpp]: If you construct an alt map like you did before and then use `merge` it is much more efficient.

Comment (ddh0) [reply common/sampling.cpp]: Addressed in `7ab9ed9`

**Review by CISC — APPROVED** [review]

**Comment by CISC** [comment]
@ngxson Final approve and merge? :)

**Review by ngxson — APPROVED** [review]
